packages/apollo-engine-reporting/src/plugin.ts

```diff
@@ -162,7 +185,28 @@ export const plugin = <TContext>(
       let didResolveSource: boolean = false;
 
       return {
+        async didResolveOperation(requestContext) {
+          if (typeof options.traceReporting !== 'function') return;
+          const shouldReportTrace = await options.traceReporting(
```

Comment:
So this doesn't happen until after parsing and validation succeed.  I don't know if this is intentional, but the docs should probably at least say that if a function is provided then things like parse and validation failures are treated as if the function returns true. Or we can make the function be invoked in that case too, eg have didEnd call the function if it wasn't called yet?

(Sure would be nice to finish up #3465...)

Reply:
We wanted it after validation and parsing so client and operation name information would be included in the request context, since that seems like useful information to use. I'll add it to the docs.

Reply:
Given that traceReporting is a brand new callback, we could easily add a second argument with flags like `parseError: boolean`, `validationError: boolean`, etc, so that users could decide in their callbacks how to handle that?

Honestly it's frustrating that #3465 stalled because that already added explicit understanding of the "big three" pre-execution error types (parse failure, validation failure, unknown operation) to AER... Real shame to add more features that ought to have access to that without it...